internal/testutils/db.go

```diff
@@ -199,3 +202,79 @@ func predicatableFieldValues(name string, data, cols []string) {
 		}
 	}
 }
+
+// PrepareDBsForTests initializes a cache in the specified directory and load it with the specified dump.
+func PrepareDBsForTests(t *testing.T, cacheDir, initialCache string, options ...cache.Option) {
+	t.Helper()
+
+	// Gets the path to the testutils package.
+	_, p, _, _ := runtime.Caller(0)
+	testutilsPath := filepath.Dir(p)
+
+	options = append(options, cache.WithTeardownDuration(0))
+
+	c := NewCacheForTests(t, cacheDir, options...)
+	err := c.Close(context.Background())
+	require.NoError(t, err, "Cache must be closed to enable the dump loading.")
+
+	for _, db := range []string{"passwd.db", "shadow.db"} {
+		LoadDumpIntoDB(t, filepath.Join(testutilsPath, "cache_dumps", initialCache, db+".dump"), filepath.Join(cacheDir, db))
+	}
+}
+
+// NewCacheForTests returns a cache that is closed automatically, with permissions set to current user.
+func NewCacheForTests(t *testing.T, cacheDir string, options ...cache.Option) (c *cache.Cache) {
+	t.Helper()
+
+	uid, gid := GetCurrentUIDGID(t)
+	opts := append([]cache.Option{}, cache.WithCacheDir(cacheDir),
+		cache.WithRootUID(uid), cache.WithRootGID(gid), cache.WithShadowGID(gid))
```

Comment:
Actually, we can set WithTearDownDuration to 0 here, and then, only in tests when it’s not 0, we can append another value to it, which should suppress previous options setting. Wdyt?

That way, the default settings for the test function makes sense.

Reply:
That was the intention... but I've ended up adding the option in the wrong function (my bad).